inference-engine/src/transformations/src/transformations/common_optimizations/wrap_interpolate_into_transposes.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include "transformations/common_optimizations/wrap_interpolate_into_transposes.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <numeric>
+#include <tuple>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include <ngraph/rt_info.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+
+namespace {
+bool transformation_is_applicable(const std::shared_ptr<ov::opset8::Interpolate>& interpolate) {
+    if (interpolate->get_input_partial_shape(0).rank().is_dynamic() || interpolate->inputs().size() != 4) return false;
+
+    int64_t input_rank = interpolate->get_input_partial_shape(0).rank().get_length();
+    // If the input rank is equal to 1 or 2, then such Interpolate is supportedby MKLDNN.
+    if (input_rank < 3) return false;
+
+    auto axes_node = std::dynamic_pointer_cast<ov::opset8::Constant>(interpolate->input_value(3).get_node_shared_ptr());
+    if (!axes_node) return false;
+
+    const auto axes = axes_node->cast_vector<int64_t>();
```

Comment:
You get this axes in two places. Please optimize the numbers of cast_vector calls.

Reply:
Done.